src/components/Image/index.tsx

```diff
@@ -37,29 +37,59 @@ function Image({source: propsSource, isAuthTokenRequired = false, session, onLoa
         },
         [onLoad, updateAspectRatio],
     );
+
+    /**
+     * trying to figure out if the current session is expired or fresh from a necessary reauthentication
+     */
+    const previousSessionAge = useRef<number | undefined>();
+    const validSessionAge: number | undefined = useMemo(() => {
+        if (session?.creationDate) {
+            if (previousSessionAge.current) {
+                if (Math.abs(previousSessionAge.current - session.creationDate) < 60000) {
+                    return session.creationDate;
+                }
+                return previousSessionAge.current;
+            }
+            if (Math.abs(new Date().getTime() - session.creationDate) >= CONST.SESSIONS_MAXIDLE_NB_HOURS * 3600000) {
+                return new Date().getTime();
+            }
+            return session.creationDate;
+        }
+        return undefined;
+    }, [session]);
+    useEffect(() => {
+        previousSessionAge.current = validSessionAge;
+    });
+
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      */
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     const source = useMemo(() => {
         if (typeof propsSource === 'object' && 'uri' in propsSource) {
             if (typeof propsSource.uri === 'number') {
                 return propsSource.uri;
             }
             const authToken = session?.encryptedAuthToken ?? null;
             if (isAuthTokenRequired && authToken) {
-                return {
-                    ...propsSource,
-                    headers: {
-                        [CONST.CHAT_ATTACHMENT_TOKEN_KEY]: authToken,
-                    },
-                };
+                if (!!session?.creationDate && new Date().getTime() - session.creationDate < CONST.SESSIONS_MAXIDLE_NB_HOURS * 3600000) {
```

Comment:
Can we create a function called `isValidSessionCreationDate` to reuse this logic?
ie:
```javascript
function isValidSessionCreationDate() {
 return !!session?.creationDate && (new Date().getTime() - session.creationDate) > CONST.SESSIONS_MAXIDLE_NB_HOURS
}
```

Reply:
i'll see how can isolate that logic in a function with a relevant name, good idea